lib/add-abort-signal.js

```diff
@@ -0,0 +1,20 @@
+export default addAbortSignal;
+
+const AbortController = window.AbortController;
+const RUNTIME_HAS_ABORTCONTROLLER = Boolean(AbortController);
+
+/**
+ * Returns a new options object, with a `signal` property added, when the
+ * runtime supports the AbortController
+ *
+ * @param  {object} options existing fetch options
+ * @return {object}      improved fetch options
+ */
+function addAbortSignal(options) {
+  const signalOptions = RUNTIME_HAS_ABORTCONTROLLER && {
+    // eslint-disable-next-line compat/compat
+    signal: new AbortController().signal
+  };
+
+  return Object.assign({}, signalOptions, options);
```

Comment:
I think signal Options should be last to force it on right?

Reply:
Not a bad idea. That will prevent the user from destroying that option by accident. I've pushed another commit that does this